src/lib/mongoBackend/mongoUpdateSubscription.cpp

```diff
@@ -457,6 +458,55 @@ static void setLastNotification(const BSONObj& subOrig, CachedSubscription* subC
 }
 
 
+
+/* ****************************************************************************
+*
+* setLastFailure -
+*/
+static long long setLastFailure(const BSONObj& subOrig, CachedSubscription* subCacheP, BSONObjBuilder* b)
```

Comment:
Can't say I really remember either. timesFailed and lastFailure are 'copies' of 'count' and 'lastNotification'. That is probably reason number one for these functions. I do however have a slight memory of a function test that failed and this fixed the problem.

We might have to spend some time to think this over.